main.go

```diff
@@ -30,42 +31,69 @@ func init() {
 	httpcaddyfile.RegisterHandlerDirective("dynamic_proxy", parseCaddyfile)
 }
 
+type AuthSites map[string]AuthSite
+
+func (a *AuthSites) Decode(input string) error {
+	*a = make(AuthSites)
+	sites := strings.Split(input, ";")
+	for _, site := range sites {
```

Comment:
Maybe make this `for _, s := ... `
Or, if you prefer, in line 44 change the first `site` to something else, so that you don't assign a new value to the `for` variable.